src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,280 @@ def inner(tree, path, root_path):
+    text_file_folder.mkdir(parents=True, exist_ok=True)
+
+    _write_api_tree_file(api_objects=api_objects, text=True)
+    _write_api_tree_file(api_objects=api_tui_objects, tui=True)
+    _write_api_tree_file(api_objects=api_object_names, name=True)
+
+
+def _process_wildcards(word: str, names: list):
+    """Process wildcard pattern in the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    wildcard_matches: list
+        Matched API object names.
+    """
+    return [name for name in names if fnmatch.fnmatch(name, word)]
+
+
+def _process_misspelled(
+    word: str,
+    names: list,
+    match_whole_word: Optional[bool] = False,
+    match_case: Optional[bool] = False,
+):
+    """Process misspelled word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+    match_whole_word: bool
+        Whether to match whole case.
+    match_case: bool
+        Whether to match case.
+
+    Returns
+    -------
+    correct_spell: list
+        List of corrected spell.
+    """
+    correct_spell = []
+    possible_corrections = []
+    spell = SpellChecker()
+    spell.word_frequency.load_words(names)
+    misspelled = spell.unknown([word])
+    if misspelled:
+        for name in misspelled:
+            correct_spell.append(spell.correction(name))
+            possible_corrections.extend(list(spell.candidates(name)))
+        if match_whole_word:
+            return correct_spell
+        elif match_case:
+            corrections_in_tree = set()
+            for correction in possible_corrections:
+                for name in names:
+                    if correction in name:
+                        corrections_in_tree.add(correction)
+            return list(corrections_in_tree)
+    else:
+        return [word]
+
+
+def _download_nltk_data():
+    """Download NLTK data on-demand."""
+    from pathlib import Path
+
+    import nltk
+
+    import ansys.fluent.core as pyfluent
+
+    nltk_data_path = Path(os.path.join(pyfluent.CODEGEN_OUTDIR, "nltk_data")).resolve()
+    nltk.data.path.append(nltk_data_path)
+    os.environ["NLTK_DATA"] = str(nltk_data_path)
+    package_path = Path(os.path.join(nltk_data_path, "corpora"))
+    wordnet = package_path / "wordnet.zip"
+    omw = package_path / "omw-1.4.zip"
+    packages = ["wordnet", "omw-1.4"]
+    for package in packages:
+        if not (package_path / f"{package}.zip").exists():
+            try:
+                nltk.download(
+                    package,
+                    download_dir=nltk_data_path,
+                    quiet=True,
+                    raise_on_error=True,
+                )
+            except BaseException:
+                return False
+    if wordnet.exists() and omw.exists():
+        return True
```

Comment:
This function downloads required package files if they are not already downloaded. 

We also set `NLTK_DATA` env var to `generated/nltk_data` folder.

Based on return values we determine whether to use semantic search. 

 If package files are downloaded or already available then it returns True, then we perform semantic search.

If it encounters any errors during downloading or internet is unavailable at the user end then returns False so we skip semantic search and do the regular search using `_process_misspelled()`